internal/query/user.go

```diff
@@ -655,7 +655,10 @@ func (q *Queries) searchUsers(ctx context.Context, queries *UserSearchQueries, f
 		UserInstanceIDCol.identifier(): authz.GetInstance(ctx).InstanceID(),
 	})
 	if permissionCheckV2 {
-		query = wherePermittedOrgsOrCurrentUser(ctx, query, filterOrgIds, UserResourceOwnerCol.identifier(), UserIDCol.identifier(), domain.PermissionUserRead)
+		query, err = wherePermittedOrgsOrCurrentUser(ctx, query, filterOrgIds, UserResourceOwnerCol.identifier(), UserIDCol.identifier(), domain.PermissionUserRead)
+		if err != nil {
+			return nil, zerrors.ThrowInternal(err, "AUTHZ-HS4us", "Errors.Internal")
```

Comment:
1. I believe the internal error is now thrown twice. Once in `wherePermittedOrgsOrCurrentUser` and again here with the parent wrapped. Please keep only one of them. Make you choice will be consistent in `wherePermittedOrgs`
2. For the error ID, use `QUERY-<random string>`. `AUTHZ` is typically used by the `authz` package I would say.